### ISSUE TITLE
Rename mentions of Redhawk to NativeAOT.

### ISSUE THREAD

**Issue by teo-tsirpanis** [dotnet/runtime#69115]
Similar to what #68950 and #68888 have done with CoreRT, we could rename mentions of Redhawk to NativeAOT. I see 1182 occurences of it in this repo.

**Comment by prasanta10** [comment]
Can i

> Similar to what #68950 and #68888 have done with CoreRT, we could rename mentions of Redhawk to NativeAOT. I see 1182 occurences of it in this repo.

Can i ?_

**Comment by huoyaoyuan** [comment]
And remember to add it to historical glossary to explain the `Rh` prefix.

**Comment by prasanta10** [comment]
Sure. I have started working on it.

**Comment by prasanta10** [comment]
> And remember to add it to historical glossary to explain the `Rh` prefix.
can i work on it?

**Comment by teo-tsirpanis** [comment]
Sure @prasanta10, go ahead.

**Comment by danmoseley** [comment]
Cc @jkotas . Are there places where this would not be a good rename?

**Comment by prasanta10** [comment]
#69187

**Comment by jkotas** [comment]
We should be able to remove all mentions of redhawk. 

The simple rename does not always work well in comments, the comment is sometimes completely obsolete, etc.

I would recommend doing this in two phases. Submit a simple find&replace of the most frequently used terms in one PR (e.g. `FEATURE_REDHAWK` -> `FEATURE_NATIVEAOT`, `REDHAWK_API` -> `NATIVEAOT_API`) and do the remaining renames that require manual tweaks in second PR once the first one is merged.

> the Rh prefix.

The Rh prefix can be go away too. Again, I would recommend to do that in a separate PR and have some upfront discussion about the naming pattern to use to eliminate the Rh prefix.

**Comment by prasanta10** [comment]
> We should be able to remove all mentions of redhawk.
> 
> The simple rename does not always work well in comments, the comment is sometimes completely obsolete, etc.
> 
> I would recommend doing this in two phases. Submit a simple find&replace of the most frequently used terms in one PR (e.g. `FEATURE_REDHAWK` -> `FEATURE_NATIVEAOT`, `REDHAWK_API` -> `NATIVEAOT_API`) and do the remaining renames that require manual tweaks in second PR once the first one is merged.
> 
> > the Rh prefix.
> 
> The Rh prefix can be go away too. Again, I would recommend to do that in a separate PR and have some upfront discussion about the naming pattern to use to eliminate the Rh prefix.

Thank you for the advice,Sir. I will be the doing the same that you told me.

**Comment by prasanta10** [comment]
#69227

**Comment by danmoseley** [comment]
@prasanta10 do you plan to continue ? no obligation.
